### DISCUSSION TITLE
Transformed Cart line_item. No item_components field as the docs say

### DISCUSSION THREAD

**Discussion by AlexanderLindkjaer** [Shopify/function-examples#335] [Feedback: Cart Transform API]
Hey :)

I got the cart transform working, both merging and expanding, But i do not se the bundle component data anywhere other than checkout.

The bundle [docs](https://shopify.dev/docs/apps/selling-strategies/bundles#lifecycle-of-a-bundle) says "Cart page shows bundle parent, with the option to return bundle components using [item_components](https://shopify.dev/docs/api/liquid/objects/line_item#line_item-item_components)."

And the liquid objects docs also clearly says the [item_components](https://shopify.dev/docs/api/liquid/objects/line_item#line_item-item_components) field should be there. 

But the field is not present anywhere in the cart. Neither in the liquid Cart object or through the api. 
The bundle line item has a field called `has_components` that is true. I cant find any reference to this field in the docs.
![screenshot 43](https://github.com/Shopify/function-examples/assets/33224052/f72c4328-14fd-420f-9666-6313a0bbb02a)

Is the `item_components` field not working or am i missing something?

**Comment by ignacio-chiazzo** [comment]
Hi, thanks for reporting. 

>Is the item_components field not working or am i missing something?

To clarify, do you mean the [Liquid API](https://shopify.dev/docs/api/liquid/objects/line_item#line_item-item_components) field `item_components`

**Reply by AlexanderLindkjaer** [reply]
Yes!

I cant get it "to return bundle components using [item_components](https://shopify.dev/docs/api/liquid/objects/line_item#line_item-item_components)" as the docs says.

**Reply by siddhantbajaj** [reply]
Hey, can you share the liquid code where you are accessing the `item_components`? Or if you have a test store with the complete setup, can you share the link to that store?

**Reply by AlexanderLindkjaer** [reply]
This suddenly started working in the meantime. I promise it wasn't before 😅 

For anybody looking, this gives access to bundle item_components in liquid

```liquid
{%- for item in cart.items -%}
 {%- for item_component in item.item_components -%}
   <p>{{ item_component.id }}</p>
 {%- endfor -%}
{%- endfor -%}
```


Feel free to close. Thanks guys.

**Comment by Tekniqu3** [comment]
It would be if item_components include in ajax api too (store_url/cart.js)

**Reply by andrewchen5678** [reply]
Same here, right now it just shows "has_components": true or false but it doesn't show the actual components

**Comment by RromeroLG** [comment]
Is it possible to show this on the product page? so we can see display each item.name and item.quantity associated to each variant?
